src/modules/iotjs_module_buffer.c

```diff
@@ -411,12 +409,12 @@ JHANDLER_FUNCTION(Slice) {
   int length = end - start;
   IOTJS_ASSERT(length >= 0);
 
-  iotjs_jval_t jnew_buffer = iotjs_bufferwrap_create_buffer(length);
+  iotjs_jval_t jnew_buffer = iotjs_bufferwrap_create_buffer((size_t)length);
   iotjs_bufferwrap_t* new_buffer_wrap =
       iotjs_bufferwrap_from_jbuffer(&jnew_buffer);
   iotjs_bufferwrap_copy_internal(new_buffer_wrap,
-                                 iotjs_bufferwrap_buffer(buffer_wrap), start,
-                                 end, 0);
+                                 iotjs_bufferwrap_buffer(buffer_wrap),
+                                 (size_t)start, (size_t)end, 0);
```

Comment:
`bound_range` is always called with `0` (as a low range). If we change the index from int to size_t, than the index cannot be lower than `0`. But we still have to check that before we call `bound_range`.  So we basically moving out the low range check from `bound_range` and copy everywhere in this file. That means a lot of code duplication which could lead errors later if we must change the code. IMHO it is not a good idea.